tests/test_optimization/test_task31_hot_path.py

```
@@ -0,0 +1,383 @@
+        assert not any("COUNT(*)" in s for s in executed), \
+            f"COUNT ran on the non-empty branch: {executed}"
+
+    def test_unanchored_candidates_closed_when_anchors_exist(self, recording_em):
+        em, executed = recording_em
+        _insert_anchor(em, 99)           # anchor exists, but not for candidate 1
+        survivors = em.filter_by_query_similarity("a query", [1], "nl")
+        assert survivors == set()
+        assert any("COUNT(*)" in s for s in executed)
+
+    def test_cold_start_fails_open(self, recording_em):
+        em, executed = recording_em      # no anchors at all
```

Ruff (0.16.1)

[warning] Unpacked variable `executed` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)